cmd/kubectl.go

```diff
@@ -37,34 +48,66 @@ func runKubectl(ctx *cli.Context) error {
 		return err
 	}
 
-	cluster, err := getClusterByID(c, c.UserConfig.FocusedCluster())
+	config, err := loadConfig(ctx)
 	if err != nil {
 		return err
 	}
 
-	config, err := c.ManagementClient.Cluster.ActionGenerateKubeconfig(cluster)
-	if err != nil {
-		return err
+	currentRancherServer := config.FocusedServer()
+	if currentRancherServer == nil {
+		return fmt.Errorf("no focused server")
 	}
 
-	tmpfile, err := ioutil.TempFile("", "rancher-")
+	currentToken := currentRancherServer.AccessKey
+	t, err := c.ManagementClient.Token.ByID(currentToken)
 	if err != nil {
 		return err
 	}
-	defer os.Remove(tmpfile.Name())
 
-	_, err = tmpfile.Write([]byte(config.Config))
+	currentUser := t.UserID
+	currentCluster := currentRancherServer.FocusedCluster()
+	filename := fmt.Sprintf("%s-%s-kubeconfig", currentUser, currentCluster)
```

Comment:
Should this format be a constant at the top of the file?

Reply:
The format is only used once so it could be but is there a benefit?

Reply:
Hmm not sure. I like to keep these as constants on the top even if it's used once since I don't like magic strings, but I'm fine with either decision!